integrations/gitlab/gitlab_integration/ocean.py

```diff
@@ -21,7 +21,16 @@
 async def handle_webhook(group_id: str, request: Request) -> dict[str, Any]:
     event_id = f'{request.headers.get("X-Gitlab-Event")}:{group_id}'
     body = await request.json()
-    await event_handler.notify(event_id, group_id, body)
+    await event_handler.notify(event_id, body)
+    return {"ok": True}
+
+
+@ocean.router.post("/system/hook")
+async def handle_system_webhook(request: Request) -> dict[str, Any]:
+    body = await request.json()
+    # some system hooks have event_type instead of event_name in the body, such as merge_request events
+    event_name = body.get("event_name") or body.get("event_type")
+    await system_event_handler.notify(event_name, body)
```

Comment:
Because it is now possible not to setup those listeners you might have an error if a request will come to the system hook route when the system hook setup_listener was not called

When notifying the handler looks for the event in a dict (handlers[event]) and in python if you try to access a values that does not exists in the list there is an exception

Maybe we should use `.get` for the handlers or disable the hooks route

the same applied to the regular hooks